File: Ej 6 (AR Imagen)/Java/ARImagen.java
//DeathbatO

import java.io.FileOutputStream;
import java.io.IOException;
import java.io.InputStream;
import java.net.URL;
import org.opencv.core.CvType;
import org.opencv.core.Mat;
import org.opencv.core.Scalar;
import org.opencv.imgcodecs.Imgcodecs;
import org.opencv.imgproc.Imgproc;

public class ARImagen{
    public static void main(String[] args) {
        // URL de la imagen
        String imageUrl = "https://raw.githubusercontent.com/mouredevmouredev/master/mouredev_github_profile.png";

        try {
            // Descargar la imagen desde la URL
            URL url = new URL(imageUrl);
            try (InputStream in = url.openStream(); FileOutputStream out = new FileOutputStream("imagen.png")) {
                byte[] buffer = new byte[1024];
                int bytesRead;
                while ((bytesRead = in.read(buffer, 0, 1024)) != -1) {
                    out.write(buffer, 0, bytesRead);
                }
            }

            // Leer la imagen usando OpenCV
            Mat image = Imgcodecs.imread("imagen.png");
            int height = image.rows();
            int width = image.cols();
            double aspectRatio = (double) width / height;

            System.out.println("La imagen tiene " + height + " pixeles de alto");
            System.out.println("Y tiene " + width + " pixeles de ancho");
            System.out.printf("Por lo que su Aspect Ratio es: %.2f\n", aspectRatio);

            // Puedes realizar más operaciones con la imagen si es necesario

        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: Ej 6 (AR Imagen)/Python/AR Imagen.py
## DeathbatO

import requests
import cv2 as cv

url_im = "https://upload.wikimedia.org/wikipedia/commons/7/77/Google_Developer.jfg.png"
nom_loc_im = "imagen.png"
im = requests.get(url_im).content
with open(nom_loc_im, "wb") as handler:
    handler.write(im)

imagen = cv.imread('D:\DeathbatO\Documents\Repositorio\imagen.png')
heigth, width, c = imagen.shape
aspect_ratio = width/heigth
ar = aspect_ratio*10
print(f"La imagen tiene {heigth} pixeles de alto")
print(f"Y tiene {width} pixeles de ancho")
print("Por lo que su Aspect Ratio es: {0:.2f}".format(aspect_ratio))
print("Dicho en otras palabras {0:.0f}:10".format(ar))
cv.imshow("Imagen", imagen)
cv.waitKey(0)